### DISCUSSION TITLE
Bump to .NET 7 causes ValueUpdate and ValueNotification events to not fire

### DISCUSSION THREAD

**Discussion by georgeinva2004** [zwave-js/ZWaveJS.NET#27] [General]
Not quite sure if this is a bug or not.  I have a .NET 6 app that works as expected.  I bumped it to .NET 7 and the Node Value Update and Node Value Notification events do not fire.  The Node Statistics update and controller events do fire.  Took a peek at the code and I don't see anything obvious that would match up with any of the breaking changes introduced in .NET 7.  Any ideas?

**Comment by georgeinva2004** [comment]
One more thing to add.  I downloaded the latest code from GitHub and set some breakpoints to determine where the problem may be.  The WatsonWsClient only appears to be receiving the Node Statistics Update event from the server.  So, it may be something with the underlying  .NET Websocket library.

**Reply by marcus-j-davies** [reply]
HI @georgeinva2004 

> The WatsonWsClient only appears to be receiving the Node Statistics

This makes me believe it's not the websocket - the websocket doesn't  care what's been received -  it will just push it through the framework to userland - I can't imagine the body of the payload will make much difference  - its all JSON events.

Have you recently built your own PSI?, or has this literally been since .NET 7?
I have (Evan now) not been impressed with websocket support in .NET - I am tempted to go about it a different way.

can you try and do a `GetValue` request? - this will tell us if the socket is sending out other things, and not just Node Statistics.

**Comment by georgeinva2004** [comment]
Okay.  Not a bug!!!!  I'm running both the .NET 6 and .NET 7 apps out of visual studio and using the debugger.  When .NET 5/6/7 apps get compiled, the binaries get put into a folder called .net5.0/.net6.0/.net7.0 respectively.  My theory is that when the Server.PSI gets downloaded it either creates a new local database or attempts to open an existing one.  A new database requires that Server.PSI re-interview all devices on the connected Z-Wave Controller.  The devices I am working with are wireless and were sleeping, so no interview was completed.   I pressed the button on each of the devices several time and I got an Interview Completed event, and voila, the ValueUpdate and ValueNotification events started firing.   Yup, I probably should have known better.  Live and learn.

**Reply by marcus-j-davies** [reply]
🤣 

That makes complete sense.

I Instruct the driver (embedded in the PSI) to use relative paths.

https://github.com/zwave-js/ZWaveJS.NET/blob/ffb4f89a15ae97ad5baf7aa262800bdbde1cef6a/Visual%20Studio%20Projects/ZWaveJS.NET/ZWaveJS.NET/ZWaveOptions.cs#L114

**Reply by marcus-j-davies** [reply]
You can override that In the `ZWaveOptions`  class instance

**Reply by georgeinva2004** [reply]
Good to know!!!!  So far so good on .NET 7.

**Reply by marcus-j-davies** [reply]
I'm still tempted to move away from the websocket approach - I find it a little fragile sometimes.
I also need to update the PSI image to bring it upto date with the latest driver.

Once time allows, I'll dig into it.
